server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -214,38 +194,34 @@ static void addCollectorsAndSearch(SearchContext searchContext) throws QueryPhas
                 );
             }
 
-            QueryPhaseCollector.CollectorManager queryPhaseCollectorManager = QueryPhaseCollector.createManager(
-                topDocsCollectorManager,
+            CollectorManager<Collector, QueryPhaseResult> collectorManager = QueryPhaseCollectorManager.createQueryPhaseCollectorManager(
                 postFilterWeight,
-                searchContext.terminateAfter(),
                 aggsCollectorManager,
-                searchContext.minimumScore()
+                searchContext,
+                hasFilterCollector
             );
```

Comment:
All of the wrapping is no longer necessary. Collectors are still wrapped when necessary, but there's a single collector manager taking care of that, including profiling.